### PR TITLE
Add tests for general GDI+ apis and implement missing APIs

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: tests/testicocodec.c [hunk 1]
```diff
@@ -82,16 +82,14 @@ static void test_invalidImage ()
 int
 main (int argc, char**argv)
 {
-  GdiplusStartupInput gdiplusStartupInput;
-  ULONG_PTR gdiplusToken;
-  GdiplusStartup (&gdiplusToken, &gdiplusStartupInput, NULL);
+	STARTUP;
```
Comment (filipnavara) [inline tests/testicocodec.c]: Wrong indentation.

File: tests/testhelpers.h [hunk 2]
```diff
@@ -9,6 +10,17 @@
 #include <unistd.h>
 #endif
 
+#define STARTUP \
+    ULONG_PTR gdiplusToken; \
+	GdiplusStartupInput gdiplusStartupInput; \
```
Comment (filipnavara) [inline tests/testhelpers.h]: Wrong indentation.

File: tests/testhatchbrush.c [hunk 3]
```diff
@@ -154,16 +154,14 @@ static void test_getBackgroundColor ()
 int
 main (int argc, char**argv)
 {
-    GdiplusStartupInput gdiplusStartupInput;
-    ULONG_PTR gdiplusToken;
-    GdiplusStartup (&gdiplusToken, &gdiplusStartupInput, NULL);
+	STARTUP;
```
Comment (filipnavara) [inline tests/testhatchbrush.c]: Wrong indentation.

File: tests/testbrush.c [hunk 4]
```diff
@@ -74,14 +74,12 @@ static void test_delete ()
...
-	ULONG_PTR gdiplusToken;
-	GdiplusStartup (&gdiplusToken, &gdiplusStartupInput, NULL);
+	STARTUP;
 
     test_getBrushType ();
     test_clone ();
     test_delete ();
 
-	GdiplusShutdown (gdiplusToken);
+    SHUTDOWN;
```
Comment (filipnavara) [inline tests/testbrush.c]: Wrong indentation.

Comment (filipnavara) [reply tests/testbrush.c]: Eh, this one is actually fine. :-)

**Comment by akoeplinger** [comment]
The indentation errors are all in test files which already use spaces (and are thus "wrong") so I'm going to just merge this as is to speed up the process.

**Pull request merged**